### PR TITLE
🐛 Fix .getImpl race condition on Android

### PR DESCRIPTION
**Pull request by mszylkowski**
Fixes https://github.com/ampproject/error-reporting/issues/13

GetImpl is being called without waiting for the custom element to load so sometimes the getImpl doesn't exist for the story.

### CODE REVIEW

**Review by Enriqe — APPROVED** [review]

File: extensions/amp-story/1.0/amp-story-page.js [hunk 1]
```diff
@@ -409,12 +409,14 @@ export class AmpStoryPage extends AMP.BaseElement {
...
-      (reason) => this.mediaPoolRejectFn_(reason)
-    );
+    whenUpgradedToCustomElement(storyEl).then(() => {
+      storyEl.getImpl().then(
+        (storyImpl) => {
+          this.mediaPoolResolveFn_(MediaPool.for(storyImpl));
+        },
+        (reason) => this.mediaPoolRejectFn_(reason)
+      );
+    });
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js:419]: Looks good but we should use promise chaining vs nesting in the future

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/test/test-amp-story-grid-layer.js [hunk 2]
```diff
@@ -57,7 +57,9 @@ describes.realWin('amp-story-grid-layer', {amp: true}, (env) => {
     });
 
     const story = win.document.createElement('amp-story');
+    // Makes whenUpgradedToCustomElement() resolve immediately.
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/test-amp-story-grid-layer.js:60]: I think this line belongs on top of `story.createdCallback = Promise.resolve();`

**Review by Enriqe — APPROVED** [review]

File: extensions/amp-story/1.0/test/test-amp-story-page.js [hunk 3]
```diff
@@ -72,7 +72,9 @@ describes.realWin('amp-story-page', {amp: {extensions}}, (env) => {
     });
 
     const story = win.document.createElement('amp-story');
+    // Makes whenUpgradedToCustomElement() resolve immediately.
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/test-amp-story-page.js:75]: ditto

**Comment by mszylkowski** [comment]
@gmajoulet passing the tests without faking the promises for amp-story requires a lot of setup to make the amp-story load correctly (which is not the point of the `test-amp-story-page.js/test-amp-story-grid-layer.js` unit test). The tests on those files don't really require anything from `amp-story` (as it should be) so I think resolving the promise is the most straightforward solution to make the tests pass without introducing overhead.

**Review by gmajoulet — APPROVED** [review]

**Pull request merged**